sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -536,6 +537,13 @@ public Read withMaxCapacityPerShard(Integer maxCapacity) {
       return toBuilder().setMaxCapacityPerShard(maxCapacity).build();
     }
 
+    /**
+     * Returns a {@link PTransform} for PCollection of {@link byte[]}, dropping Kinesis metatdata.
+     */
+    public PTransform<PBegin, PCollection<byte[]>> withoutMetadata() {
+      return new TypedWithoutMetadata(this);
+    }
```

Comment:
I think it will be confusing to have this function here, since every other `with...` method returns a `Read` class. A user might expect to be able to do:

```
KinesisIO.read()
   .withoutMetadata()
   .withMaxCapacityPerShard(x)
```

Instead we should add a `KinesisIO.readWithoutMetadata()` that returns something like `TypedWithoutMetadata` with all the same configuration parameters as `Read`. You might take a look at how this is handled in `PubsubIO` for inspiration, I think it has a similar problem where it supports multiple different output types, but we want them all to share some configuration parameters. It looks like the way it's handled there is with a Read<T> type, and you pass in a function for converting PubsubMessage to T.

Alternatively, you could move the logic to extract the payload into the external transform builder, rather than adding the interface here (I think there's value in adding it here though if you want to take it on).

Reply:
Thanks for pointing it, I followed the KafkaIO convention and didn't think whether it's a good design or not. The change is trivial, just use KinesisIO.readWithoutMetadata() and pass Read there instead of creating it in withoutMetadata(this).

Reply:
I'll leave it in KinesisIO.